genequery-rest/src/main/kotlin/gq/rest/services/OverlapService.kt

```diff
@@ -0,0 +1,52 @@
+package gq.rest.services
+
+import gq.core.data.GQModule
+import gq.core.data.Species
+import gq.core.data.intersectWithSorted
+import gq.core.genes.GeneFormat
+import gq.rest.GQDataRepository
+import gq.rest.exceptions.BadRequestException
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+
+data class OverlapResponse(val identifiedGeneFormat: String,
+                           val geneConversionMap: Map<String, Long?>,
+                           val overlapGenes: LongArray)
+
+@Service
+open class OverlapService @Autowired constructor(
+        private val gqDataRepository: GQDataRepository) {
+
+    open fun convertGenes(rawGenes: List<String>,
```

Comment:
Duplicated code from [GeneSetEnrichmentService](https://github.com/ctlab/genequery-kotlin/blob/master/genequery-rest/src/main/kotlin/gq/rest/services/GeneSetEnrichmentService.kt#L23) class. This code should be reused.